platforms/intellij/compose/src/jvmMain/kotlin/foundry/intellij/compose/aibot/LoadingAnimation.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2024 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package foundry.intellij.compose.aibot
+
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.LinearOutSlowInEasing
+import androidx.compose.animation.core.RepeatMode
+import androidx.compose.animation.core.infiniteRepeatable
+import androidx.compose.animation.core.keyframes
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.delay
+import org.jetbrains.jewel.foundation.theme.JewelTheme
+
+// tutorial from https://www.youtube.com/watch?v=xakNOVaYLAg
+
+@Composable
+fun LoadingAnimation(
+  modifier: Modifier = Modifier,
+  dotSize: Dp = 7.dp,
+  dotColor: Color = JewelTheme.contentColor,
+  spacing: Dp = 7.dp,
+  movementDistance: Dp = 10.dp,
+) {
+  val animatedDots = List(4) { remember { Animatable(0f) } }
```

Comment:
This will re-allocated every time, shouldn't the remember be outside of the `List` creation?

Reply:
Yeah, you are right. The remember should be outside the List block because otherwise it would be calling remember on each item of the list. Thank you for catching that, I made the change here: 3ca30396d163bb823843057e0ba62aea5a66f24c